content/ADR-162-auth-chain-with-permissions.md

```diff
@@ -0,0 +1,311 @@
+scene deployment in Genesis City it can be the coordinates of the land, or
+for a profile deployment it can be the profile address.
+
+The `allow` and `deny` elements are structured as follows:
+`namespace:service-name:operation-name` or `namespace:service-name:*`.
+The namespace is used to for prefixing all the services that belong to an
+organization. For e.g., in Decentraland the prefix could be `dcl`. The
+second part, the service name, refers to the service that this permission
+declaration is intended for. It has to be unique for services within the
+same namespace. The last part can be either a single operation name or a `*`
+character meaning any possible operation for the service.
+
+In the example above, resource `0xaddress` allows `dcl:explorer` to do any
+operations except `denied-operation`. A single operation name always weighs
+more than the wildcard operator, and hence takes precedence over it.
+
+### Permission readability
+
+One of the key points in the permission specification is that they must be
+human-readable, because the user signing the creation of the ephemeral key needs
+to be able to read it in plain text and understand what he is about to sign.
+
+On the other hand, the text has to be machine parsable so that the server is
+able to parse the message signed by the user that contains the permission
+specification and be able to interpret it for allowing or disallowing the
+operation request accordingly.
+
+Here is a few options proposed for permission specification. Options 1 and 2
+propose different payload options to be
+signed, based
+on [EIP-4361 Sign-In with Ethereum](https://eips.ethereum.org/EIPS/eip-4361).
+
+Option 1: more yaml-like:
+
+```text
+Decentraland Login
+Ephemeral address: 0x4f9c4Ff265357F937B14BCd1E5519f1537985301
+Expiration: 2022-12-29T00:00:00.000Z
+
+Permissions:
+  - resource: "menduz.dcl.eth"
+    allow:    "dcl:worlds:deploy"
+  - resource: "menduz.dcl.eth"
+    allow:    "unicorn:worlds:deploy"
+  - resource: "0,0"
+    allow:    "dcl:scene:deploy"
+  - resource: "0xaddress"
+    allow:    "dcl:explorer:*"
+```
+
+Option 2: yaml-like list of parsable phrases:
+
+```text
+Decentraland Login
+Ephemeral address: 0x4f9c4Ff265357F937B14BCd1E5519f1537985301
+Expiration: 2022-12-29T00:00:00.000Z
+
+Permissions:
+- allow "dcl:worlds:deploy" for menduz.dcl.eth
+- allow "dcl:explorer:*" for 0xaddress
+- deny "dcl:explorer:voice" for 0xaddress
+- deny "dcl:explorer:voice2" for 0xaddress
+- allow "unicorn:worlds:deploy" for menduz.dcl.eth
+- allow "dcl:scene:deploy" for *
+```
+
+Option 3: Typed structured data
+
+An alternative approach to permission specification would be to use structured
+data signing, as defined in
+[EIP-712 Typed structured data hashing and signing](https://eips.ethereum.org/EIPS/eip-712).
+
+The follow snippet can be run in the browser console:
+
+```javascript
+let domain = [
+    {name: "name", type: "string"},
+    {name: "chainId", type: "uint256"}
```

Comment:
chainId is not mandatory. If we pass the `chainId` the users won't be able to sign the message if they are in a different network. This is the same case when user signs meta transactions for Polygon by being connected to Ethereum. We use the [`salt` parameter to specify the chainId](https://github.com/decentraland/wearables-contracts/blob/master/contracts/commons/EIP712Base.sol#L16). 

I may suggest adding the version here as well.

About the salt, we can use it if we want to prevent ambiguity for some reason

Reply:
Thanks Nacho. This snipped of code came from some example I got on the web, we definitely need to revise the details, so whatever you guys suggest is fine.
This doc has a lot to be updated. The main part that needs revision is the way we express the permissions so that it is valid for the use cases we have today. And apart from being usable from backend side, it also has to be decently presented for end users who have to sign the ephemeral key. 
@fzavalia was going to try out a few ways to organize the permission data to see if we can decide on a way that covers both sides (clear to the user that's signing the key and expressive enough for common scenarios in the explorer or other websites).